domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java

```diff
@@ -190,14 +180,41 @@ private static SecurityDomain toSecurityDomain(final FutureTask<SecurityDomain>
         }
     }
 
+    private static FutureTask<SecurityDomain> toFutureTask(InjectedValue<SecurityDomain> injectedValue) {
+        return new FutureTask<>((Callable<SecurityDomain>)  (() -> injectedValue.getValue()));
+    }
+
     static class IdentityService implements Service<Void> {
 
         private List<FutureTask<SecurityDomain>> getSecurityDomainFutures = Collections.emptyList();
+        private final ManagementSecurityIdentitySupplier securityIdentitySupplier;
+
+        public IdentityService(ManagementSecurityIdentitySupplier securityIdentitySupplier) {
+            this.securityIdentitySupplier = securityIdentitySupplier;
+        }
 
         @Override
         public void start(StartContext context) throws StartException {
             if (getSecurityDomainFutures != null) {
-                getSecurityDomainFutures.forEach(FutureTask::run);
+                Runnable task = () -> {
+                    try {
+                        getSecurityDomainFutures.forEach(FutureTask::run);
+                        List<Supplier<SecurityDomain>> domains = getSecurityDomainFutures.stream().map(f -> (Supplier<SecurityDomain>) (() -> toSecurityDomain(f))).collect(Collectors.toList());
+                        securityIdentitySupplier.setConfiguredSecurityDomainSupplier(domains.get(0));
+                        domains.remove(0);
+                        if (domains.size() > 0) {
+                            securityIdentitySupplier.setInflowSecurityDomainSuppliers(domains);
+                        }
+                        context.complete();
+                    } catch (RuntimeException t) {
+                        context.failed(DomainManagementLogger.SECURITY_LOGGER.unableToConfigureManagementSecurityDomain(t));
+                    }
+                };
+                try {
+                    task.run();
+                } finally {
+                    context.asynchronous();
```

Comment:
This wasn't proper use of context.asynchronous, because 'task' isn't handed off to a different thread.

But, good news is, 'task' isn't needed nor are the context.asynchronous() / context.complete() calls. There's nothing blocking going on here, just a bunch of calls to injectedValue.getValue().